templates/commands/goldentest/test_funcs_test.go

```diff
@@ -584,7 +584,69 @@ steps:
 			}
 
 			gotDestContents := common.LoadDirWithoutMode(t, filepath.Join(tempDir, "testdata/golden/test"))
-			if diff := cmp.Diff(gotDestContents, tc.want, common.CmpFileMode); diff != "" {
+			if diff := cmp.Diff(gotDestContents, tc.want); diff != "" {
+				t.Errorf("dest directory contents were not as expected (-got,+want): %s", diff)
+			}
+		})
+	}
+}
+
+func TestRenameGitDirsAndFiles(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name         string
+		filesContent map[string]string
+		want         map[string]string
+		wantErr      string
+	}{
+		{
+			name: "simple_success",
+			filesContent: map[string]string{
+				".git/config": "gitconfig contents",
+				".git/ref":    "gitref contents",
+				".gitignore":  "gitignore contents",
+				"file1.txt":   "file1",
+			},
+			want: map[string]string{
+				".git.abc_renamed/config": "gitconfig contents",
+				".git.abc_renamed/ref":    "gitref contents",
+				".gitignore.abc_renamed":  "gitignore contents",
+				"file1.txt":               "file1",
+			},
+		},
+		{
+			name: "non_root_gitignore_success",
+			filesContent: map[string]string{
+				"subfolder1/.gitignore": "subfolder1 gitignore contents",
+				"subfolder2/.gitignore": "subfolder2 gitignore contents",
+				"file1.txt":             "file1",
+			},
+			want: map[string]string{
+				"subfolder1/.gitignore.abc_renamed": "subfolder1 gitignore contents",
+				"subfolder2/.gitignore.abc_renamed": "subfolder2 gitignore contents",
+				"file1.txt":                         "file1",
+			},
+		},
+	}
```

Comment:
Perhaps also have a test case where a directory is renamed containing a file that is also renamed?